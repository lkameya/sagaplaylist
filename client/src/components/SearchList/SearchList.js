import React from 'react';
import styles from './SearchList.module.css';

const SearchList = props => (
  <div className={styles.searchList}>
    <ul className={styles.list}>
      {props.songs.map(song => {
        return (
          <div className={styles.track} onClick={() => props.addToPlay(song)}>
            <img src={song.album.images[1].url} alt="albumimg" className={styles.albumImage} height={70} />
            <li
              className={styles.listItem}
              key={song.id}
            >{song.name} - {song.artists[0].name}
            </li>
          </div>
        )
      })}
    </ul>
  </div>
)

export default SearchList;
