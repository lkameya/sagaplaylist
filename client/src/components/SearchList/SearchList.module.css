.searchList {
  height: 200px;

}

.list {
  display: flex;
  justify-content: center;
  list-style: none;
  font-size: 12px;
  margin-top: 50px;
}

.listItem {
  margin: 10px 5px;
}

.track {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 150px;
  text-align: center;
}

.track:hover {
  cursor: pointer;
  transform: scale(1.1);
  transition: .2s all;
}
.track:active {
  transform: translateY(-10px);
}

.albumImage {
  border-radius: 150px;
  margin-bottom: 5px;
}