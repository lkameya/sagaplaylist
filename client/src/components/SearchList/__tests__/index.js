import React from 'react';
import SearchList from '../SearchList';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

const mock = {
  addToPlay: () => '',
  songs: [{
    id: '1',
    album: {
      images: [{},
      {
        url: 'Teste'
      }]
    },
    artists: [
      {
        name: 'SOAD'
      }
    ]
  }]
}

describe('Search List', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SearchList {...mock}>
      </SearchList>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});