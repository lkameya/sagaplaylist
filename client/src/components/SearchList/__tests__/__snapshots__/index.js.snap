// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search List renders correctly 1`] = `
.c0 {
  height: 25vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  list-style: none;
  font-size: 1.2rem;
  margin-top: 5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 15rem;
  text-align: center;
}

.c2:hover {
  cursor: pointer;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition: .2s all;
  transition: .2s all;
}

.c2:active {
  -webkit-transform: translateY(-1rem);
  -ms-transform: translateY(-1rem);
  transform: translateY(-1rem);
}

.c3 {
  border-radius: 15rem;
  margin-bottom: .5rem;
  height: 8rem;
}

.c4 {
  margin: 1rem .5rem;
}

@media only screen and (max-width:37.5em) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

@media only screen and (max-width:37.5em) {
  .c3 {
    height: 6rem;
  }
}

<div
  className="c0"
>
  <ul
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <img
        alt="albumimg"
        className="c3"
        src="Teste"
      />
      <li
        className="c4"
      >
         - 
        SOAD
      </li>
    </div>
  </ul>
</div>
`;
